test/e2e/service.go

```diff
@@ -797,6 +797,75 @@ var _ = Describe("Services", func() {
 	})
 })
 
+var _ = Describe("Services", func() {
+	f := NewFramework("services-temp")
+
+	var t *explorerTest
+	BeforeEach(func() {
+		t = newDefaultExplorerTest(f.Client, f.Namespace.Name)
+	})
+
+	It("should serve round-robin services round-robin", func() {
+		_, _, svc := t.setup()
+
+		nodeName := getFirstNodeNameOrDie(f.Client)
+
+		h := histogram{}
+		var lastHostname string
+		sampleSize := 20
+		for i := 0; i < sampleSize; i++ {
+			// we have to ssh in order to view the cluster through the lense of a Pod,
+			// i.e. this is the easiest way to only proxy through a single kube-proxy.
+			// unfortunately we can't use the apiserver proxy for this.
+			stdout, stderr, code, err := SSH("curl --silent http://"+svc.Spec.ClusterIP+"/hostname/", nodeName, testContext.Provider)
+			Expect(err).NotTo(HaveOccurred(), "error executing ssh: %v", err)
+			Expect(code == 0).To(BeTrue(), "error code was expected to be 0, but was %d: %s", code, stderr)
+
+			hostname := strings.Trim(stdout, " ")
+			Expect(hostname == lastHostname).To(BeFalse(), "this hostname should not have been the last hostname")
+			h.observe(hostname)
+		}
+		for _, count := range h {
```

Comment:
In it's current form, I think that this test only works correctly for strictly 2-pod services?
Best to either generalize it to n-pod services, or clearly comment it's limitation.